VirtualReality/Widgets/qMRMLVirtualRealityView.cxx

```diff
@@ -515,6 +520,52 @@ void qMRMLVirtualRealityViewPrivate::updateTransformNodeWithControllerPose(vtkEv
   node->EndModify(disabledModify);
 }
 
+//----------------------------------------------------------------------------
+void qMRMLVirtualRealityViewPrivate::updateTransformNodeWithHMDPose()
+{
+  vtkMRMLLinearTransformNode* node = this->MRMLVirtualRealityViewNode->GetHMDTransformNode();
+
+  if (node == NULL)
+  {
+    qCritical() << "Unable to retrieve linear transform node for HMD";
+    return;
+  }
+
+  int disabledModify = node->StartModify();
+  if (this->RenderWindow->GetTrackedDeviceIndexForDevice(vtkEventDataDevice::HeadMountedDisplay) == vr::k_unTrackedDeviceIndexInvalid)
+  {
+    node->SetAttribute("VirtualReality.HMDActive", "0");
+    node->EndModify(disabledModify);
+    return;
+  }
+
+  vr::TrackedDevicePose_t& pose = this->RenderWindow->GetTrackedDevicePose(this->RenderWindow->GetTrackedDeviceIndexForDevice(vtkEventDataDevice::HeadMountedDisplay));
+  if (pose.eTrackingResult != vr::TrackingResult_Running_OK)
+  {
+    node->SetAttribute("VirtualReality.HMDActive", "0");
+  }
+  else
+  {
+    node->SetAttribute("VirtualReality.HMDActive", "1");
+  }
+
+  double pos[3];
```

Comment:
pos, ppos, wxyz, wdir may be left uninitialized depending on what happens in ConvertPoseToWorldCoordinates. It would be nice to explicitly initialize these variables to make the behavior more predictable.

double pos[3]={0.};
double ppos[3]={0.};
double wxyz[4]={1., 0., 0., 0.};
double wdir[3]={1., 0., 0.};